### PR TITLE
Remove third_party/upb/BUILD

### PR DESCRIPTION
**Pull request by veblush**
This is not necessary to build and may interfere with other tools because those can be confused with alien BUILD file.

### CODE REVIEW

**Comment by markdroth** [comment]
It seems like it's going to be a pain to remember to delete this file every time we pull in a new version of upb.  Can you put together a script to update upb that takes care of this automatically?

**Comment by veblush** [comment]
@roth Yes, I'm thinking of having a script after your upb PR is merged, also we will have Nico's PR (#22151) to prevent the repo from having alien BUILD file accidentally, which indirectly helps too.

**Comment by veblush** [comment]
@nicolasnoble I ran into this error (sigh). It was working on my workstation. Do you have an idea why it happened?

```
+ /var/local/git/grpc/tools/bazel build @com_google_protobuf//:protoc
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 1.0.0 will be used instead of system-wide bazel installation.
...
/root/.cache/bazel/_bazel_root/d2dc70c3d9da3fab488ba0dcbbd35051/external/com_google_protobuf/BUILD:427:1: 
Linking of rule '@com_google_protobuf//:protoc' failed (Exit 1) clang failed: error executing command /usr/bin/clang @bazel-out/k8-fastbuild/bin/external/com_google_protobuf/protoc-2.params
Use --sandbox_debug to see verbose messages from the sandbox
clang: error: invalid linker name in argument '-fuse-ld=/usr/bin/ld.gold'
Target @com_google_protobuf//:protoc failed to build
```

**Comment by veblush** [comment]
It seems that the docker base image that our sanity docker image is based upon became a bit stale. The version of clang in the image is 3.8 which doesn't support that option. So I'm now trying to upgrade the base docker image to the latest.

File: templates/tools/dockerfile/test/cxx_sanitizers_jessie_x64/Dockerfile.template [hunk 1]
```diff
@@ -15,7 +15,7 @@
   # limitations under the License.
 
   # This is the base Docker image we use for running tests on RBE
-  FROM gcr.io/cloud-marketplace/google/rbe-debian8@sha256:1ede2a929b44d629ec5abe86eee6d7ffea1d5a4d247489a8867d46cfde3e38bd
+  FROM gcr.io/cloud-marketplace/google/rbe-debian8@sha256:cda3a8608d0fc545dffc6c68f6cfab8eda280c7a1558bde0753ed2e8e3006224
```
Comment (jtattermusch) [inline templates/tools/dockerfile/test/cxx_sanitizers_jessie_x64/Dockerfile.template]: btw, this docker file is only used for run_tests.py sanitizers and it seems like we don't need them anymore
(and we are not even running them).
So at some point we're probably going to delete this dockerfile.

**Comment by veblush** [comment]
To make a script simpler, I changed it to use `debian:10` instead of `debian:9` which requires a sort of hack to download the clang v5. But the `debian:10` frees us from using that with the newer version of clang. Let's see how it works well.

File: templates/tools/dockerfile/test/cxx_sanitizers_jessie_x64/Dockerfile.template (hunk 1 above)
Comment (veblush) [reply templates/tools/dockerfile/test/cxx_sanitizers_jessie_x64/Dockerfile.template]: It turned out that I changed the wrong file. As you said, it doesn't seem to be used. I can try to remove this file.

**Comment by veblush** [comment]
New sanity image seems to work but it complains more because it gets smarter. Since it needs some work to take a look at, I'm creating another PR only for upgrading docker image. Good thing is there is no more `check_upb_output.sh` breakage, meaning this PR will be merged after that.

**Comment by markdroth** [comment]
There are actually still a bunch of issues that Josh and Nico are still helping to resolve with the upb upgrade, so I'm not sure how soon it will be merged.  I'll keep you posted.

**Pull request merged**